templates/kops-kubectl-config.sh.j2

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+set -e
+set -u
+set -o pipefail
+
+NAME="{{ cluster.name }}"
+export KOPS_STATE_STORE="s3://{{ cluster.s3_bucket_name }}"
+
+RUN=0
+{% raw %}
+if [ "${#}" -gt "0" ] && [ "${1}" = "--yes" ]; then
+{% endraw %}
+	RUN=1
+fi
+
+{# Start of AWS Account check #}
+{% if kops_default_aws_account_limit | length > 0 %}
+if [ "${RUN}" -eq "1" ]; then
+	if ! command -v aws >/dev/null 2>&1; then
+		>&2 echo "Error, aws binary not available but required to check for correct AWS account."
+		exit 1
+	fi
+	if ! AWS_ACCOUNT="$( aws sts get-caller-identity --output text --query 'Account' )"; then
```

Comment:
this is not profile aware, yet 😄

Reply:
And after second thought it should not be.  Script is triggered independently of Ansible and there is no way to tell if profile is used or not.